drivers/src/dma.rs

```diff
@@ -543,70 +537,89 @@ impl<'a> DmaRecovery<'a> {
         block_size: u32,
         write_addr: AxiAddr,
         fixed_addr: bool,
+        offset: u32,
     ) -> CaliptraResult<()> {
         self.dma.flush();
 
         let write_transaction = DmaWriteTransaction {
             write_addr,
             fixed_addr,
             length: payload_len_bytes,
-            origin: DmaWriteOrigin::Mbox,
+            origin: DmaWriteOrigin::Mbox(offset),
         };
-        self.dma.setup_dma_write(write_transaction);
-        self.dma.set_block_size(block_size);
-        self.dma.do_transaction()?;
+        self.dma.setup_dma_write(write_transaction, block_size);
+        self.dma.wait_for_dma_complete();
         Ok(())
     }
 
     // Downloads an image from the recovery interface to the mailbox SRAM.
     pub fn download_image_to_mbox(&self, fw_image_index: u32) -> CaliptraResult<u32> {
-        const INDIRECT_FIFO_DATA_OFFSET: u32 = 0x68;
-        const RECOVERY_DMA_BLOCK_SIZE_BYTES: u32 = 256;
-
         self.with_regs_mut(|regs_mut| {
             let recovery = regs_mut.sec_fw_recovery_if();
 
-            // Set PROT_CAP:Byte11 Bit3 (i.e. DWORD2:Bit27) to 1 ('Flashless boot').
-            recovery.prot_cap_0().modify(|val| val | (1 << 27));
+            // Set PROT_CAP2 Bit27 to 1 ('Flashless boot').
+            recovery
+                .prot_cap_2()
+                .modify(|val| val | (1 << Self::PROT_CAP2_FLASHLESS_BOOT_BIT));
 
             // Set DEVICE_STATUS:Byte0 to 0x3 ('Recovery mode - ready to accept recovery image').
-            // Set DEVICE_STATUS:Byte[2:3] to 0x12 ('Recovery Reason Codes' 0x12 = 0 Flashless/Streaming Boot (FSB)).
-            recovery
-                .device_status_0()
-                .modify(|val| (val & 0xFF00FF00) | (0x12 << 16) | 0x03);
+            // Set DEVICE_STATUS:Byte[2:3] to 0x12 ('Recovery Reason Codes' 0x12 - Flashless/Streaming Boot (FSB)).
+            recovery.device_status_0().modify(|val| {
+                (val & Self::RESET_BYTE_0_2_3_MASK)
+                    | (Self::FLASHLESS_STREAMING_BOOT_VALUE << 16)
+                    | Self::READY_TO_ACCEPT_RECOVERY_IMAGE_VALUE
+            });
 
-            // Set RECOVERY_STATUS:Byte0 Bit[3:0] to 0x1 ('Awaiting recovery image') &
-            // Byte0 Bit[7:4] to 0 (Recovery image index).
-            // [TODO][CAP2] the spec says this register is read-only, but there is no other way to select an image index?
+            // Set Byte0 Bit[3:0] to 0x1 ('Awaiting recovery image')
+            // Set Byte0 Bit[7:4] to recovery image index
             recovery.recovery_status().modify(|recovery_status_val| {
-                // Set Byte0 Bit[3:0] to 0x1 ('Awaiting recovery image')
-                // Set Byte0 Bit[7:4] to recovery image index
-                (recovery_status_val & 0xFFFFFF00) | (fw_image_index << 4) | 0x1
+                (recovery_status_val & Self::RESET_BYTE_0_MASK)
+                    | (fw_image_index << Self::FW_IMG_INDEX_BIT_POS)
```

Comment:
I feel like we're hard-coding a lot more offsets here, but the updated RDL has a lot of these defined: https://github.com/chipsalliance/i3c-core/blob/dc3afa3df7a6dfc992d9bcbc1db8a52fe6695011/src/rdl/secure_firmware_recovery_interface.rdl#L468

Could we update the RDL and regenerate the registers, and use the autogenerated offsets instead (where possible)?

Reply:
(and the `.set()`, `etc.` methods.